packages/parser/src/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { JSXElement, createMemo } from "solid-js";
+
+export function createParser(id?: string) {
+  const $TOKEN = Symbol(id || "solid-parser");
+
+  function tokenize<
+    TToken extends Meta & {
+      props: unknown;
+    }
+  >(meta: Omit<TToken, "props">): (props: TToken["props"]) => JSXElement {
+    return (props: any) => {
+      return Object.assign(
+        () => {
+          console.info("tokens can only be rendered inside a Parser");
+          return <></>;
+        },
```

Comment:
This could be the default rendering method (just warn), but it could probably be customizable. You might have a component that is valid to use inside and outside the parser.
Although I would mostly parser-only so I like the default

Reply:
I like that! An optional second parameter in tokenize ===  the component used when u try to render the token directly?